Framework/DataHandling/src/LoadDetectorInfo.cpp

```diff
@@ -274,7 +280,8 @@ void LoadDetectorInfo::loadFromIsisNXS(const std::string &filename) {
     double pressure = detInfo.pressures[i];
     double thickness = detInfo.thicknesses[i];
 
-    updateParameterMap(pmap, det, l2, theta, phi, delta, pressure, thickness);
+    updateParameterMap(pmap, detectorInfo.detector(index), l2, theta, phi,
```

Comment:
Maybe we could think about making this nicer as we shrink the responsibilities of `Detector`